mta-multi-tenant.yaml

```diff
@@ -53,83 +64,107 @@ modules:
       commands:
         - npm run build
       requires:
-        - name: bookshop-mt-srv
+        - name: bookshop-mt-ias-srv
     requires:
-      - name: bookshop-mt-service-manager
-      - name: bookshop-mt-uaa
+      - name: bookshop-service-manager
+      - name: bookshop-identity
+        parameters:
+          config:
+            credential-type: "X509_GENERATED"
+            key-length: 2048
+            validity: 30
+            validity-type: "DAYS"
+            app-identifier: "microservice1"
       - name: cf-logging
     provides:
       - name: mtx-api
         properties:
           mtx-url: ${default-url}
-# --------------------- APPROUTER MODULE ---------------------
-  - name: bookshop-mt-app
-# ------------------------------------------------------------
+  # --------------------- APPROUTER MODULE ---------------------
+  - name: bookshop-mt-ias-app
+    # ------------------------------------------------------------
     type: approuter.nodejs
     path: app
     parameters:
       memory: 256M
       disk-quota: 512M
       keep-existing-routes: true
     properties:
-      TENANT_HOST_PATTERN: ^(.*)-${default-uri} # testing only, use custom domain with wildcard for production
+      TENANT_HOST_PATTERN: ^(.*)-${default-host}.${default-domain} # testing only, use custom domain with wildcard for production
     requires:
-    - name: srv-api
-      group: destinations
-      properties:
-        name: backend
-        url: ~{srv-url}
-        forwardAuthToken: true
-        strictSSL: true
-    - name: bookshop-mt-uaa
+      - name: srv-api
+        group: destinations
+        properties:
+          name: backend
+          url: ~{srv-cert-url}
+          forwardAuthCertificates: true
+          forwardAuthToken: true
+          strictSSL: true
+      - name: bookshop-identity
+        parameters:
+          config:
+            credential-type: "X509_GENERATED"
+            key-length: 2048
+            validity: 30
+            validity-type: "DAYS"
+            app-identifier: "microservice1"
+      - name: bookshop-sms
     provides:
       - name: app-api
         properties:
           app-url: '${default-url}'
-          app-domain: '${domain}'
+          app-domain: '${default-domain}'
 # --------------------- RESOURCES ---------------------
 resources:
-# -----------------------------------------------------
-  - name: bookshop-mt-uaa
-    type: org.cloudfoundry.managed-service
-    parameters:
-      service: xsuaa
-      service-plan: application
-      path: ./xs-security-mt.json
-      config: # override xsappname as it needs to be unique
-        xsappname: bookshop-mt-${org}-${space}
-        oauth2-configuration:
-          redirect-uris:
-          - https://*.~{app-api/app-domain}/**
-    requires:
-      - name: app-api
-  - name: bookshop-mt-service-manager
+  # -----------------------------------------------------
+  - name: bookshop-service-manager
```

Comment:
from a naming pattern perspective, we chose a name for service instances which makes it easy to correlate to the scenario (e.g. as integration tests for all scenarios are deployed into the same subaccount). What about `bookshop-mt-ias-service-manager`?

https://github.com/SAP-samples/cloud-cap-samples-java/blob/messaging/mta-multi-tenant-messaging.yaml

Reply:
I was wondering whether that's necessary in general, or whether it's possible to share these service instances.

But I guess, regarding the service manager, you'd want a dedicated one. But what about the identity service?

Reply:
As you said, it depends on the service type. Service manager might also be reusable, but to avoid interferences between the different test scenarios, I wouldn´t do it. Identity service should not cause issues if we share service instance. Howver, why not just adding something like `...-mt-ias` to all service instances. This would make it easier to manage our integration test subaccount and have consistency with service instance naming.